pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -6,48 +6,93 @@
 package configresolver
 
 import (
-	"bytes"
 	"context"
 	"errors"
 	"fmt"
+	"net/url"
 	"os"
+	"regexp"
 	"strconv"
+	"strings"
 
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/listeners"
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
 	"github.com/DataDog/datadog-agent/pkg/util/containers"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
-	"github.com/DataDog/datadog-agent/pkg/util/tmplvar"
+
+	yaml "gopkg.in/yaml.v2"
 )
 
-type variableGetter func(ctx context.Context, key []byte, svc listeners.Service) ([]byte, error)
+type variableGetter func(ctx context.Context, key string, svc listeners.Service) (string, error)
 
 var templateVariables = map[string]variableGetter{
 	"host":     getHost,
 	"pid":      getPid,
 	"port":     getPort,
 	"hostname": getHostname,
+	"env":      getEnvvar,
 	"extra":    getAdditionalTplVariables,
 	"kube":     getAdditionalTplVariables,
 }
 
+type fakeService struct{}
```

Comment:
maybe `fake` is not a good name. I don't have a clear name, maybe: `conversionService` `intermediateService`

Reply:
I’ve just completely removed it and replaced its use by use of `nil` instead.